parquet/src/main/java/org/apache/iceberg/parquet/ParquetVariantWriters.java

```diff
@@ -98,6 +99,26 @@ static ParquetValueWriter<VariantValue> objects(
         builder.build());
   }
 
+  public static ParquetValueWriter<?> array(
+      int valueDefinitionLevel,
+      ParquetValueWriter<?> valueWriter,
+      int typedDefinitionLevel,
+      int repeatedDefinitionLevel,
+      int repeatedRepetitionLevel,
+      ParquetValueWriter<?> elementWriter) {
+    ArrayWriter typedWriter =
+        new ArrayWriter(
```

Comment:
Like with the read path, should this construct the value writer and the array writer separately?

Reply:
Make sense.